// avatar.proto
syntax = "proto3";

message AvatarPredictionMessage {
    BlendShapes blendShapes = 1; 
    Rotation rotation = 2;
    Transform transform = 3;
}

message BlendShapes {
  // Alphabetical
  float  browDown_L = 1;
  float  browDown_R = 2;
  float  browInnerUp = 3;
  float  browOuterUp_L = 4;
  float  browOuterUp_R = 5;
  float  cheekSquint_L = 6;
  float  cheekSquint_R = 7;
  float  eyeBlink_L = 8;
  float  eyeBlink_R = 9;
  float  eyeLookDown_L = 10;
  float  eyeLookDown_R = 11;
  float  eyeLookIn_L = 12;
  float  eyeLookIn_R = 13;
  float  eyeLookOut_L = 14;
  float  eyeLookOut_R = 15;
  float  eyeLookUp_L = 16;
  float  eyeLookUp_R = 17;
  float  eyeSquint_L = 18;
  float  eyeSquint_R = 19;
  float  eyeWide_L = 20;
  float  eyeWide_R = 21;
  float  jawLeft = 22;
  float  jawOpen = 23;
  float  jawRight = 24;
  float  mouthClose = 25;
  float  mouthDimple_L = 26;
  float  mouthDimple_R = 27;
  float  mouthFrown_L = 28;
  float  mouthFrown_R = 29;
  float  mouthFunnel = 30;
  float  mouthLeft = 31;
  float  mouthLowerDown_L = 32;
  float  mouthLowerDown_R = 33;
  float  mouthPress_L = 34;
  float  mouthPress_R = 35;
  float  mouthPucker = 36;
  float  mouthRight = 37;
  float  mouthRollLower = 38;
  float  mouthRollUpper = 39;
  float  mouthShrugLower = 40;
  float  mouthShrugUpper = 41;
  float  mouthSmile_L = 42;
  float  mouthSmile_R = 43;
  float  mouthStretch_L = 44;
  float  mouthStretch_R = 45;
  float  mouthUpperUp_L = 46;
  float  mouthUpperUp_R = 47;
  float  noseSneer_L = 48;
  float  noseSneer_R = 49;

   // These are not given by our models yet, so we will always set their value to 0 for now.
  float  cheekPuff = 50;
  float  jawForward = 51;
  float  tongueOut = 52;

}

message Rotation {
  optional float pitch = 1;
  optional float roll  = 2;
  optional float yaw   = 3;
}

message Transform {
  optional float x = 1;
  optional float y = 2;
  optional float z = 3;
}
